### ISSUE TITLE
[ui] Replace single redeploy request with update + deploy requests

### ISSUE THREAD

**Issue by richard-cox** [epinio/ui#277]
- Blocked on https://github.com/epinio/epinio/issues/2363
- The redeploy api request will be removed
- We need to replace this with two requests now. update and deploy
- Affects two places (need to check for other places)
  - adding / removing instances using the + / - buttons
    - any requests here should be debounced such that if the user clicks multiple times we only deploy once they have stopped. Split out to https://github.com/epinio/ui/issues/287
  - editing application source
- before starting we should discuss offering the user the choice on whether they want to deploy the app following the changes. 
  - this could be a checkbox in the edit app source flow somewhere

**Comment by enrichman** [comment]
We don't need the two API calls because in the end I've kept the original behavior adding a flag to skip the restart, so we just need to provide this flag during the push.

See: https://github.com/epinio/epinio/issues/2363#issuecomment-1633928534

**Comment by richard-cox** [comment]
@torchiaf unfortunately we don't have the same bot from dashboard running here, so we need to make sure issues that are automatically closed by PRs are reopened and moved to test

**Comment by mmartin24** [comment]
@richard-cox could you please give some hints on how to test this?
thanks

**Comment by torchiaf** [comment]
@mmartin24 This will fix this [issue](https://github.com/epinio/epinio/issues/2363).

Area that should be covered by tested:

- Edit a faulty app .
- Scale applications (regressions check).

The last point in the description is actually not covered, possibly we may want to address it in a new issue.

**Comment by mmartin24** [comment]
Checked in `Epinio Server Version: v1.9.0-26-g21ef05bd` with latest Epinio UI.

- Checked faulty up can be sourced to a working one.
- Checked once sourced to a valid one, it can scaled up and down 3 instances 

https://github.com/epinio/ui/assets/37271841/c3825b69-e9df-4a1b-8896-cd6ccba0f764

(Testing note: use `make appchart` when deploying testing cluster if local)

**Comment by mmartin24** [comment]
> @mmartin24 This will fix this [issue](https://github.com/epinio/epinio/issues/2363).
> 
> Area that should be covered by tested:
> 
> * Edit a faulty app .
> * Scale applications (regressions check).
> 
> The last point in the description is actually not covered, possibly we may want to address it in a new issue.

Thanks @torchiaf .
Tested accordingly and working.
I think the last point you mentioned regarding application scaling is covered here: https://github.com/epinio/ui/issues/287

Closing issue
